src/OpenTelemetry.Api/Context/Propagation/AWSXRayPropagator.cs

```diff
@@ -0,0 +1,276 @@
+                }
+
+                var parentHeader = parentTraceHeader.First();
+
+                if (!TryParse(parentHeader, out var rootTraceId, out var parentId, out var sampleDecision))
+                {
+                    return context;
+                }
+
+                var rootId = ExtractRootTraceId(rootTraceId);
+                var traceId = ActivityTraceId.CreateFromString(rootId.AsSpan());
+                var spanId = ActivitySpanId.CreateFromString(parentId.AsSpan());
+                var traceOptions = sampleDecision == SampledValue ? ActivityTraceFlags.Recorded : ActivityTraceFlags.None;
+
+                return new PropagationContext(new ActivityContext(traceId, spanId, traceOptions, isRemote: true), context.Baggage);
+            }
+            catch (Exception ex)
+            {
+                OpenTelemetryApiEventSource.Log.ActivityContextExtractException(nameof(AWSXRayPropagator), ex);
+            }
+
+            return context;
+        }
+
+        /// <inheritdoc/>
+        public void Inject<T>(PropagationContext context, T carrier, Action<T, string, string> setter)
+        {
+            if (context.ActivityContext.TraceId == default || context.ActivityContext.SpanId == default)
+            {
+                OpenTelemetryApiEventSource.Log.FailedToInjectActivityContext(nameof(AWSXRayPropagator), "Invalid context");
+                return;
+            }
+
+            if (carrier == null)
+            {
+                OpenTelemetryApiEventSource.Log.FailedToInjectActivityContext(nameof(AWSXRayPropagator), "null carrier");
+                return;
+            }
+
+            if (setter == null)
+            {
+                OpenTelemetryApiEventSource.Log.FailedToInjectActivityContext(nameof(AWSXRayPropagator), "null setter");
+                return;
+            }
+
+            var traceId = ToXRayTraceIdFormat(context.ActivityContext.TraceId.ToHexString());
+            var spanId = context.ActivityContext.SpanId.ToHexString();
+            var sampleDecision = (context.ActivityContext.TraceFlags & ActivityTraceFlags.Recorded) != 0 ? SampledValue : NotSampledValue;
+
+            var newTraceHeader = string.Concat(RootKey, KeyValueDelimiter, traceId, TraceHeaderDelimiter, ParentKey, KeyValueDelimiter, spanId, TraceHeaderDelimiter, SampledKey, KeyValueDelimiter, sampleDecision);
+
+            setter(carrier, AWSXRayTraceHeaderKey, newTraceHeader);
+        }
+
+        internal static bool TryParse(string rawHeader, out string rootTraceId, out string parentId, out char sampleDecision)
+        {
+            rootTraceId = null;
+            parentId = null;
+            sampleDecision = default;
+
+            try
+            {
+                if (string.IsNullOrEmpty(rawHeader))
+                {
+                    return false;
+                }
+
+                var keyValuePairs = rawHeader.Split(ValidSeparators, StringSplitOptions.RemoveEmptyEntries)
```

Comment:
In Java we do a more manual parse to avoid extra allocations - it's nice if a low level SDK optimizes these code paths. Might want to try it here. 

https://github.com/open-telemetry/opentelemetry-java/blob/master/extensions/trace_propagators/src/main/java/io/opentelemetry/extensions/trace/propagation/AwsXRayPropagator.java#L151

Reply:
Nice call out, will update in v2